android/shared/src/commonMain/kotlin/io/rebble/cobble/shared/ui/view/home/watches/WatchesPage.kt

```diff
@@ -0,0 +1,100 @@
+package io.rebble.cobble.shared.ui.view.home.watches
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import io.rebble.cobble.shared.ui.common.RebbleIcons
+import io.rebble.cobble.shared.ui.viewmodel.WatchesListViewModel
+
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun WatchesPage(viewModel: WatchesListViewModel = viewModel{ WatchesListViewModel() }) {
+    val selectedWatch = viewModel.selectedWatch.value
+
+    if (selectedWatch != null) {
+
+        WatchBottomSheetContent(
+                watch = selectedWatch,
+                onToggleConnection = { viewModel.toggleConnection(selectedWatch, true) },
+                onForgetWatch = { viewModel.forgetWatch(selectedWatch) },
+                onCheckForUpdates = { viewModel.checkForUpdates(selectedWatch) },
+                clearSelection = { viewModel.clearSelection() }
+        )
+    }
+
+    Column(
+            modifier = Modifier.fillMaxSize(),
+            verticalArrangement = Arrangement.Top
+    ) {
+        CenterAlignedTopAppBar(
+                colors = TopAppBarDefaults.centerAlignedTopAppBarColors(
+                        containerColor = MaterialTheme.colorScheme.primaryContainer,
+                ),
+                title = {
+                    Text(
+                            "My watches",
+                            maxLines = 1,
+                            overflow = TextOverflow.Ellipsis
+                    )
+                }
+        )
+        if (viewModel.getConnectedWatch() == null) {
```

Comment:
Ok, this is not a style comment :) 
Method `getConnectedWatch()`, which has a loop inside, will run on every recomposition of this Composable, you usually don't want it. 
The correct way to implement this functionality would be to define a property inside the view model and use `derivedStateOf` to obtain the needed object. Like so (defined in the view model):
```kotlin
// WatchesListViewModel.kt

val connectedWatch: WatchItem? by derivedStateOf {
        for (watch: WatchItem in _watches) {
            if (watch.isConnected) {
                watch
            }
        }
        null
}
```
This way, `connectedWatch` will be always up-to-date but only recomputed when `_watches` list is updated